app/lib/companies_selector.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class CompaniesSelector
+  def initialize(params)
+    @params = params
+  end
+
+  def select
+    if @params[:stock]
+      Company.where(stock: @params[:stock])
+    elsif @params[:query]
+      Company.where('name LIKE ?', "%#{@params[:query].upcase}%")
+        .or(Company.where('isin LIKE ?', "%#{@params[:query].upcase}%"))
+        .and(Company.where(condition: 'company active'))
+    else
+      Company.where(condition: 'company active')
+    end
+  end
+end
```

Comment:
Maybe something like this would work for you

```ruby
class CompaniesFetcher
  def call(params)
    relation = Company
    relation = filter_by_condition(relation, params.fetch(:condition)) if params.key?(:condition)
    relation = filter_by_stock(relation, params.fetch(:stock)) if params.key?(:stock)
    relation = filter_by_query(relation, params.fetch(:query)) if params.key?(:query)

    relation.all
  end

  private

  def filter_by_condition(relation, condition)
    relation.where(condition: condition)
  end

  def filter_by_stock(relation, stock)
    relation.where(stock: stock)
  end

  def filter_by_query(relation, query)
    pattern = "%#{query.upcase}%"

    relation.where('name LIKE ? OR isin LIKE ?', pattern, pattern)
  end
end
```

Reply:
This is very good idea, kind of this I was thinking about, the bells were ringing somewhere, but I didn't know where :)
This is very useful schema. I'm not sure yet but maybe kind of this could be use to set breadcrumbs

Reply:
I have look at breadcrumbs, I think for now I can live them like they are. It can shows the search results more detailed like 
`Home/GPW Stock/Search result` in case of passing 2 parameters `:stock` and `query`, or more detailed like
`Home/GPW Stock/Search result: :query`
but however the first option looks fine, I'm not sure if the second is a good practice because `:query` parameter could be anything.
Also I'm thinking if is a good practice to put breadcrumbs in one file and render them.

Reply:
I found Dynamic breadcrumbs with React Router, what do you think about this?

Reply:
> I found Dynamic breadcrumbs with React Router, what do you think about this?

You would have to add React first. It requires some effort and learning and basically rewriting your views to JavaScript ;) It will be good idea do learn it but it's a bigger thing and you would need to plan 2-3 weeks for it or something like that. So for sure not as part of this PR.

Reply:
I would try to keep breadcrumbs simple if possible. Maybe `Home/GPW Stock/Search result` or maybe even `Home/GPW Stock/Companies` (the same as without searching) :) But that's me.

Reply:
I will keep like it is for now there is no point to change them. Probably in the future when I learn React etc.